code/lib/codemod/src/transforms/__tests__/mdx-to-csf.test.ts

```diff
@@ -0,0 +1,412 @@
+import * as fs_ from 'node:fs';
+import { expect, test } from '@jest/globals';
+import dedent from 'ts-dedent';
+import jscodeshift, { nameToValidExport } from '../mdx-to-csf';
+
+expect.addSnapshotSerializer({
+  print: (val: any) => (typeof val === 'string' ? val : JSON.stringify(val, null, 2) ?? ''),
+  test: () => true,
+});
+
+jest.mock('node:fs');
+const fs = fs_ as jest.Mocked<typeof import('node:fs')>;
+
+beforeEach(() => {
+  fs.existsSync.mockImplementation(() => false);
+});
+
+test('rewrite import', () => {
+  const input = dedent`
+      import { Meta, Story } from '@storybook/addon-docs';
+    `;
+
+  const mdx = jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  expect(mdx).toMatchInlineSnapshot(`
+    import { Meta, Story } from '@storybook/blocks';
+    import * as FoobarStories from './Foobar.stories';
+
+  `);
+});
+
+test('drop invalid story nodes', () => {
+  const input = dedent`
+      import { Meta } from '@storybook/addon-docs';
+
+      <Meta title="Foobar" />
+      
+      <Story>No name!</Story>  
+    `;
+
+  const mdx = jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  expect(mdx).toMatchInlineSnapshot(`
+    import { Meta } from '@storybook/blocks';
+    import * as FoobarStories from './Foobar.stories';
+
+    <Meta of={FoobarStories} />
+
+
+  `);
+});
+
+test('convert story re-definition', () => {
+  const input = dedent`
+      import { Meta, Story } from '@storybook/addon-docs';
+      import { Primary } from './Foobar.stories';
+
+      <Meta title="Foobar" />
+      
+      <Story story={Primary} />
+    `;
+
+  fs.existsSync.mockImplementation((path) => path === 'Foobar.stories.js');
+
+  const mdx = jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  expect(mdx).toMatchInlineSnapshot(`
+    import { Meta, Story } from '@storybook/blocks';
+    import { Primary } from './Foobar.stories';
+    import * as Foobar_Stories from './Foobar_.stories';
+
+    <Meta of={Foobar_Stories} />
+
+    <Story of={Foobar_Stories.Primary} />
+
+  `);
+  const [csfFileName, csf] = fs.writeFileSync.mock.calls[0];
+  expect(csfFileName).toMatchInlineSnapshot(`Foobar_.stories.js`);
+  expect(csf).toMatchInlineSnapshot(`
+    import { Primary } from './Foobar.stories';
+
+    export default {
+      title: 'Foobar',
+    };
+
+    export { Primary };
+
+  `);
+});
+
+test('Keep story nodes with id', () => {
```

Comment:
Did we decide that resolving `id`s was too tricky?

Reply:
If so, would it make sense to inject a comment explaining that the `id` is deprecated?
like
```
{/* <Story id="button--primary" /> is deprecated, please migrate it to <Story of={referenceToStory} /> manually */}
```

Reply:
Maybe I was unclear, I think we should add a comment _above_ the Story, not wrap the whole Story in a comment. That way it would still work, but we would explain the user what they need to do.

Reply:
Got it, fixed it, hopefully my tree traversing logic for skipping the mutated entries is fail proof 🤞